### DISCUSSION TITLE
Indexing nodes and edges

### DISCUSSION THREAD

**Discussion by VincentLapointe-OPALRT** [networkx/networkx#5954] [Q&A] [answered]
Hi all,
Is there an elegant way of associating indexes to each node of a graph (or adding index to edges ) ? It could be stored as an attributes of a nodes/edges if needed
These indexes should ideally be consecutive and start from 0... if a node/edge is added/deleted/reorder, index should be updated accordingly.
These indexes will be used as row/column references in numpy 2d array. Simple example: I have a [**n** x **m**] matrix where **n** and **m** are the number of nodes and edges of the graph
Tx

**Comment by dschult** [comment, marked as answer]
The usual way of translating back and forth between nodes and the index of the nodes in the adjacency matrix of a graph G is:
```python
node_to_index = {node: indx for indx, node in enumerate(G)}
index_to_node = list(G)
```
Similar ideas could be tried for edges, but it doesn't seem to be as useful.  The edge ordering is not as natural in our data structure. Still, you could use the order of `G.edges` to determine the edge index.  I think it is mostly the incidence_matrix where that might be used.

**Reply by VincentLapointe-OPALRT** [reply]
Tx for your suggestion
